### PR TITLE
`SideNav` - Extract a "base" component

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This is a follow-up of https://github.com/hashicorp/design-system/pull/1304 in which I have extracted the "basic" layout functionality of the `SideNav` in a dedicated component, so that it can be used independently from the full-fledged one, to achieve basic navigations (in which only the layout wrapper is needed, not all the responsiveness/animation etc).

_The idea for this change has materialised when I was starting to write the documentation for "how to use" the sidebar navigation, and I've realised that in cases where a consumer would need just a basic navigation, the implemented `SideNav` could be an overkill (especially for the fact that there's a lot of JavaScript involved, that in the case of a basic navigation may not be necessary at all). So I decided to do test, see how the code would look like, and at the end of the process I saw that it wasn't bad at all, on the contrary: made sense. So I decided to open the PR and gather feedback from the other HDS engineers._

### :hammer_and_wrench: Detailed description

In this PR I have:
- added a `SideNav::Base` component (copy&pasted the existing `SideNav` and stripped away all the non-basic-layout stuff)
- updated the `SideNav` to use (internally) the `SideNav::Base` subcomponent
- added showcase example for `SideNav::Base` + moved `SideNav` examples in the showcase up in the page
- updated the “Component API” section of the website
- added integration tests for `SideNav::Base`

### :link: External links

- https://github.com/hashicorp/design-system/pull/1304

***

### 👀 Reviewer's checklist:

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/side-nav/partials/code/component-api.md [hunk 1]
```diff
@@ -2,7 +2,31 @@
...
+  </C.Property>
+</Doc::ComponentApi>
+
+### SideNav::Base
+
+This is the basic component (layout only).
+
 <Doc::ComponentApi as |C|>
+  <C.Property @name="<:root>" @type="named block">
+    A named block where content that lives in the "root" of the SideNav is rendered (the content should not cause a layout height).
```
Comment (KristinLBradley) [inline website/docs/components/side-nav/partials/code/component-api.md:28]: "the content should not cause a layout height" could you please clarify this part? Does it mean the content shouldn't include a fixed height or something else?

**Review by KristinLBradley — COMMENTED** [review]
Looks good. I just added a comment on one part that seemed unclear to me.

File: website/docs/components/side-nav/partials/code/component-api.md (hunk 1 above)
Comment (didoo) [reply website/docs/components/side-nav/partials/code/component-api.md:28]: Yeah, you're right @KristinLBradley, it's not clear (it's hard to explain). Let me have a new go at tomorrow, with a fresh mind, and see how to improve the description.

Comment (didoo) [reply website/docs/components/side-nav/partials/code/component-api.md]: @KristinLBradley I've made a change that removes the issue at the "root" (pun intended) :)

File: website/docs/components/side-nav/partials/code/component-api.md [hunk 2]
```diff
@@ -2,7 +2,31 @@
...
+  </C.Property>
+</Doc::ComponentApi>
+
+### SideNav::Base
+
+This is the basic component (layout only).
+
 <Doc::ComponentApi as |C|>
+  <C.Property @name="<:root>" @type="named block">
+    A named block that can be used to render content that needs to be rendered outside of the "header/body/footer" containers of the SideNav.
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/component-api.md:28]: suggestion – no strong opinion
```suggestion
    A named block for rendering content outside of the "header/body/footer" containers of the SideNav.
```

**Review by alex-ju — APPROVED** [review]
Looks great, could spot anything off!

Not part of this change, but just a note that in the 'yielded content' example we have multiple `<nav>` items, whereas in the portal we have a single element. For parity, we could have a single `Hds::SideNav::List` for the 'yielded content' (if only to show there's no difference in the end result).

**Comment by didoo** [comment]
> Looks great, could spot anything off!
👍

> Not part of this change, but just a note that in the 'yielded content' example we have multiple `<nav>` items, whereas in the portal we have a single element. For parity, we could have a single `Hds::SideNav::List` for the 'yielded content' (if only to show there's no difference in the end result).
@alex-ju it was intentional, to test what happens when/if a consumers uses two `List`s in the `<:body>`, to see how the SideNav component reacts (expecially in terms of vertical spacing).

**Pull request merged**